Add printout of BranchID (or optionally ProductID) to edmProvDump

makortel opened the pull request:
#### PR description:

Following the discussion in https://mattermost.web.cern.ch/cms-o-and-c/pl/jwfa7decrjda9jwrgepmw36woy this PR extends `edmProvDump` to print out the `BranchID` of each product. Optionally the the `BranchID` can be translated to `ProductID` with `--productIDEntry N` command line argument, where `N` is the Event entry number to do the translation.

#### PR validation:

Framework unit tests run. `edmProvDump` of a file shows BranchID (or ProductID) for products, also for dropped products.

makortel committed:
Add printout of ProductID (or BranchID) to edmProvDump

If not able to find out the ProductID, print out BranchID. The event
entry number to do the transformation can be customized with a new
command line argument.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-38858/31267

- This PR adds an extra 40KB to repository

cmsbuild commented:
A new Pull Request was created by  @makortel (Matti Kortelainen) for master.

It involves the following packages:

- DataFormats/Provenance (**core**)
- FWCore/Framework (**core**)
- IOPool/Common (**core**)


@cmsbuild, @smuzaffar, @Dr15Jones, @makortel can you please review it and eventually sign? Thanks.
@wddgit, @rovere this is something you requested to watch as well.
@perrotta, @dpiparo, @qliphy, @rappoccio you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

makortel commented:
@cmsbuild, please test

makortel commented:
@wddgit @Dr15Jones Could you double-check this makes sense?

In the `EdmProvDump.cc` I did not even attempt to support earlier file format versions (it should end up printing the BranchID in such cases). What do you think, would that be sufficient?

Dr15Jones commented:
given that the actually productID to branchID depends on exactly which event one is looking at, I would error on the side of turning off the printout and ONLY do a print if given an event index. That way it is not possible for someone to be mislead.

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ec201c/26451/summary.html
**COMMIT**: b814bfb2b86cdf6a66daa7a85542fd248748700c
**CMSSW**: CMSSW_12_5_X_2022-07-25-1100/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/38858/26451/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test TestEdmConfigDump had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 51
 - DQMHistoTests: Total histograms compared: 3667670
 - DQMHistoTests: Total failures: 13
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3667634
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 50 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 210 log files, 47 edm output root files, 51 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_5_X_2022-07-25-1100+ec201c/51799/triggerResults): no differences found

makortel commented:
Fixed the unit test, changed the default behavior to print out the BranchID, and renamed the command line parameter to `--productIDEntry` to do the BranchID-to-ProductID translation.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-38858/31282

- This PR adds an extra 40KB to repository

cmsbuild commented:
Pull request #38858 was updated. @cmsbuild, @smuzaffar, @Dr15Jones, @makortel can you please check and sign again.

makortel commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ec201c/26472/summary.html
**COMMIT**: ae6c00c94043d2aa99524a009201b24195c50e67
**CMSSW**: CMSSW_12_5_X_2022-07-26-1100/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/38858/26472/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 51
 - DQMHistoTests: Total histograms compared: 3667670
 - DQMHistoTests: Total failures: 19
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3667628
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 50 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 210 log files, 47 edm output root files, 51 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_5_X_2022-07-26-1100+ec201c/51816/triggerResults): no differences found

makortel commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @qliphy, @rappoccio (and backports should be raised in the release meeting by the corresponding L2)

qliphy commented:
+1
